src/main/java/jcmp/DocCompare.java

```diff
@@ -394,6 +393,9 @@ public int hashCode() {
 
         @Override
         public boolean equals(Object x) {
+            if (!(x instanceof DefaultIdentity)) {
```

Comment:
stack trace will be generated on failure only.

On Tue, Mar 29, 2016 at 7:58 AM, Naveen Malik <notifications@github.com>
wrote:

> In src/main/java/jcmp/DocCompare.java
> <https://github.com/bserdar/jiff/pull/1#discussion_r57726975>:
>
> > @@ -394,6 +393,9 @@ public int hashCode() {
> >
> >          @Override
> >          public boolean equals(Object x) {
> > +            if (!(x instanceof DefaultIdentity)) {
>
> There is cost with creating a stack trace. It will be faster to not rely
> on try-catch.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/bserdar/jiff/pull/1/files/4efbd8f32d48864623357249c53f773bdd917044#r57726975>
>